octopoes/tools/rename-origin-method.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+import json
+import logging
+from typing import Any
+
+import click
+from xtdb_client import XTDBClient
+
+logger = logging.getLogger(__name__)
+
+
+@click.group(
+    context_settings={
+        "help_option_names": ["-h", "--help"],
+        "max_content_width": 120,
+        "show_default": True,
+    }
+)
+@click.option("-n", "--node", default="0", help="XTDB node")
+@click.option(
+    "-u",
+    "--url",
+    default="http://localhost:3000",
+    help="XTDB server base url",
+)
+@click.option(
+    "-t",
+    "--timeout",
+    type=int,
+    default=5000,
+    help="XTDB request timeout (in ms)",
+)
+@click.option("-v", "--verbosity", count=True, help="Increase the verbosity level")
+@click.pass_context
+def cli(ctx: click.Context, url: str, node: str, timeout: int, verbosity: int):
+    verbosities = [logging.WARN, logging.INFO, logging.DEBUG]
+    try:
+        if verbosity:
+            logging.basicConfig(level=verbosities[verbosity - 1])
+    except IndexError:
+        raise click.UsageError("Invalid verbosity level (use -v, -vv, or -vvv)")
+
+    client = XTDBClient(url, node, timeout)
+    logger.info("Instantiated XTDB client with endpoint %s for node %s", url, node)
+
+    ctx.ensure_object(dict)
+    ctx.obj["client"] = client
+
+
+QY = '{:query {:find [(pull ?var [*])] :where [[?var :type "Origin"][?var :origin_type "observation"][?var :method]]}}'
+
+
+def method_query(method: str):
+    if method:
+        return f"""{{
+:query {{
+    :find [(pull ?var [*])] :where [
+            [?var :type "Origin"]
+            [?var :origin_type "observation"]
+            [?var :method "{method}"]
+        ]
+    }}
+}}
+"""
+    else:
+        return QY
+
+
+@cli.command(help="List observation origins based on method")
+@click.argument("method", required=False)
+@click.pass_context
+def LIST(ctx: click.Context, method: str):
+    origins = ctx.obj["client"].query(method_query(method))
+    if not origins:
+        raise click.UsageError("No targets found")
+    if "error" in origins:
+        raise click.UsageError(origins["error"])
+    click.echo(json.dumps(origins))
+
+
+def search_replace_method(ds: list[dict[str, Any]], s: str, r: str):
```

Comment:
Please consider using more descriptive variable names. The current names are too short and make the code unnecessarily complex. A few suggestions:
- Rename `ds` to `data_list` to indicate it is a list of dictionaries
- Rename `d` to `data_dict` to indicate it is a dictionary
- Rename `k` to `key` to indicate it is a key in the dictionary
- Rename `s` to `search_str` to indicate it is the string to search for
- Rename `r` to `replace_str` to indicate it is the string to replace with

You also need to use `None` as a return type hint

Reply:
Done in e8cc0c51673b8c0569cc630276b64db46a81f088.